### DISCUSSION TITLE
自动识别敏感话题的机制

### DISCUSSION THREAD

**Discussion by KYLN24** [OpenTreeHole/backend#111] [Ideas]
**Is your feature request related to a problem? Please describe.**
树洞内容审查完全依靠人工，难免出现极度敏感的内容处理不及时的问题。因此，需要引入自动化处理机制。

**Describe the solution you'd like**
在新内容被添加到数据库后，提交至审核器进行审核。审核器为命中的内容添加*敏感*效果。

### *敏感*效果

处于*敏感*效果下的内容将直接隐藏，在前端提示“内容审核中”，并交由管理员审核。管理员审核通过后恢复显示，如不通过，则将前端提示修改为管理员的删除理由。

### 审核器

审核器可以是独立的进程。审核器对于敏感问题的识别灵敏度应当可以调节。

文字和图片均应进行审核，可以采用自然语言处理和图像识别相关算法，亦可调用第三方接口。如调用第三方接口，应当在隐私政策中注明。

考虑到服务器性能问题，待审核的内容应当排列在任务队列中依次审核，无需立即取得结果。暂未审核的内容可以在前端正常显示。

在审核完成前已被管理员处理的内容不应当继续进行自动审核流程。

**Comment by hasbai** [comment]
运营管理员的职责是接受用户的反馈并妥善地处理它们，而不是主动地去寻找并消灭那些“错误”的内容。

如果说这样人工化的管理流程还可以被认为是为了生存迫不得已的手段的话，自动化成体系的审查和屏蔽就俨然成为了威权系统的延伸与外化。在这样的体系中，人类的思想在机器和算法面前成为了屠宰场传送带上排列有序的羔羊，等待着某种不可知的规则对其评估出一个指标，然后决定它应当存活还是被抹杀——这不是自由的匿名社区，而是互联网时代的奥斯维辛。

无论是开发者还是运营者不应该想方设法地消减，而应当尽力去捍卫言论自由——这是我们在这个时代仅存的人性与良知。

**Comment by KYLN24** [comment]
被动地管理会使社区加速地走向灭亡，因为人性在不加约束的情况下必然会倾向于诞生一切人反对一切人的战争。

一切工具都是为人服务的，这一自动化机制仍然是为确保公约的落实。工具为人的管理带来方便，最终删除某些内容依然是人工确认的结果。如果使用工具就是被工具奴役，就是人类的思想被宰割，那人类何以发展到今天？痛批着工具的人，不也是在使用着 Python、Vue、Django、Docker、GitHub 等等工具吗？更何况，如果说真的有什么是所谓的裁剪着人们言论自由的普洛克路斯忒斯之床，那不过是注册之时人人皆签署同意的公约罢了。公约本质上是人所拟定，把对公约化管理的不满迁怒到机器之上，把人的过错推卸到算法之上，这是不是对机器的霸凌，对算法的残酷呢？

如果对公约有意见，应当去完善它。如果对公约化管理体系有意见，就去提出更好的管理方式（我并不觉得被动式管理是一个好的方式）。如果承认公约化管理，那么无论是用户还是运营者都应确保公约的严格落实，否则所谓公约不过是一纸空文——这是我们在这个时代需要的共识与担当。

**Comment by w568w** [comment]
Issue of a repository is intended to make discussions about **technical details** and **feasibility of feature requests**, rather than **a place for moral criticism and debates over administrations of its implementations and community guidelines.**

If you guys have to indulge in exaggerations with non-development business, **go to Slack and talk about it.** No more here please.

@hasbai If you insists that this issue is meaningless or harmful, just *close* it directly. I will not hold my own opinion.

**Comment by hasbai** [comment]
既然你提到了“一切人反对一切人的战争”，那么我们不妨用此作一类比。霍布斯之所以提出“自然状态”的假想是为了论证国家（政府）存在的合法性。国家暴力机关存在的一个原因就是为了制止犯罪，避免产生“一切人反对一切人的战争”的状态，那么与之对应的我们论坛的管理团队的存在也正是为了履行这一职责，在管理团队的运作下，不友善的、不合法的言论将被删除，这不会让社区走向灭亡，因此你第一点的逻辑并不成立。

至于主动和被动管理的问题，在这样一个类比下，我们知道几乎所有国家的执法机构都是你所谓的“被动管理”：警察接到报案 --> 调查案件 --> 制止/惩处犯罪分子，又有几个国家的警察会乐于每天前往黑帮林立枪战频繁的街区维护正义呢？

对于工具的问题，你所列出的工具都是“良器”，这些工具的使用本身是不会作恶的。然而公权力则不然，公权力会侵犯到一些人的权利，它是作恶的，因此必须要对公权力使用的工具作出审慎的限制。设想这样一个“主动管理”的未来世界：空中遍布着搭载摄像头与监听器的无人机，随时随地监控你的一举一动，把你的行为传输给 AI 分析，一旦发现你有犯罪行为，就立即把你制服。这必然会是一个没有罪恶的世界，但没有人愿意生活在这样的世界中，不仅仅是隐私权的问题，而是在于公权力的极度强化。即使是在善意的情况下，也没有人愿意住在这样一个全景敞视监狱中，更不用说当它被恶意地使用，如果这套系统被纳粹用于种族灭绝，那当今世界一个犹太人也不会存在。

此外，我完全赞同按照公约管理的模式，我都前述观点也没有与此违背的地方。

**Comment by hasbai** [comment]
对于技术上的问题也可以讨论，我很清楚是什么原因促使你提出这个 issue，但以现有的技术能力真的能很好的实现吗？社区公约允许人们表达自己的政治观点，在这场区域战争中我可以支持某个国家也可以反对某个国家，这是很合理的。但那位用户的问题在于他将自己的政治观点与大学这样一个机构组织放到了一起，给人一种“学校也支持他的政治观点”的感觉，这种“碰瓷”学校的行为无疑是错误的。

可是 AI 如何才能发现这样的错误呢？AI 可以识别出这张图片中含有一句政治口号，也能识别出图片中有一幢学校代表建筑物，但它能认为这幢建筑物等于学校官方意志吗？它能从这两样事物中得到一种“学校支持这种政治观点”的暗示吗？它能把这种暗示匹配到某个特定的规则并对帖子是否违规进行评判吗？我认为以目前的技术无法做到这点。如果这样的技术连这个已经发生过的问题都无法正确处理，我们又如何希望它能有助于今后论坛的运营管理工作呢？

**Comment by KYLN24** [comment]
感谢你终于意识到自己把讨论错误地引向了哪里。并及时回头开始研讨技术问题。

管理团队将不友善、不合法的言论删除，避免社区走向灭亡，这就是所谓的“非被动的主动式管理”，也就是你所说的“而不是主动地去寻找并消灭那些’错误‘的内容”中的主动。这一点上你自相矛盾。因为管理员的删除并非主要来源于用户的举报。

国家机关显然不是完全放弃掉你所说的“而不是主动地去寻找并消灭那些’错误‘的内容”中的主动，公安部门的治安巡逻等就是明证。如果仅靠群众报案，那还拿公安局、纪检委、检察院来干什么呢？而且实不相瞒，你所在的国家就是愿意主动派人前往黑帮林立的地方进行扫黑除恶工作并且成绩斐然。

至于“良器”与“恶器”，我也不知道你如何作出区分。我亦不知道你将敏感预警机制等价于公权力是出于怎样的逻辑。我认为工具没有善恶，重要的是使用它的人的意图。建立一套预警机制，无论是对用户维护良好社区环境还是对运营者减轻工作负担都是有益的，无论从什么角度看来都不是“恶器”。

你所设想的赛博朋克大戏实在是精彩，但这和本议题讨论的是同一个事物吗？寻求一种预警机制，在你看来就等价于关键字识别式的言论销毁吗？首先是你所说的隐私权，树洞内的言论均暴露于全社区之下，我认为对这些内容进行识别根本不涉及任何对隐私权的侵犯。其次，预警机制进行预警之后的最终裁量依然在运营者这个不是 AI 的人手中。运营者删帖，谈何是 AI 将你制服？让 AI 背大锅 AI 岂不是太冤了？难道因为运营者没关注到或者没有及时关注到就任其流行的违法乱纪仇恨恶意言论就是你希望看到吗？工具被纳粹使用而成为恶器，你要成为纳粹吗？

我对你上述发言稍加总结，就是在我提出需要有预警措施后，你提出了一种严格暴力的、识别后不经人工确认的彻底删除机制，然后用了许多华丽的辞藻来驳斥你自己提出的方案。我不清楚这样做的意义何在，并且我不觉得你提出的这种冷血机械的实现能对管理工作帮上什么忙。

你所描述的观点，与你对按照公约的管理模式有诸多违背。

而最终，你终于开始思考技术上的问题。如果你认为预警机制在技术上不可实现，那大可直接表明，而不是经过两大段居高临下、自立标靶的道德审判后，才如此曲折地提出。诚然，目前的工具不能理解图片。但当它发现一张图片具有学校、政治、乌克兰等关键词时，如果它将这张图片即使地暂时隐藏，并通知管理员立即审核，管理员审核后确认删除，我相信就没有后续一定要由官方势力介入的诸多事宜。而我相信，减少了官方势力的介入，一定能更加保障社区的言论自由。不加约束的自由不会造就一切人反对一切人的战争那还会产生什么？如你所见，从发帖到官方主要到此内容不过十几分钟。你不盯着，有人替你盯着，只不过处理的主动权以及事态的大小就完全不是一回事了。并且，我对于科技的发展始终抱有信心，即使现有的技术难以有效实现预警机制，但这并不意味着预警机制本身的错误。设想有人在社区内教唆自杀，该帖若并不热门，管理员没有及时发现。但被教唆者切切实实地通过推送通知即使地了解了该信息，并受到了其影响。我想如果缺乏预警机制，我们管理员无法尽到对社区里的每一个个体应有的保护义务。公约服务的是一个个活生生的人，而不是用户群体这一形而上的概念。

我提议，你的描述中提出的那种你自己都不惜用大段文字驳斥的实现方式切勿采纳，而具体的更合适的实现方式有待商讨，这也正是本议题 help wanted 所需要的讨论。希望后续的讨论者着眼于如何利用所谓的“良器”使运营者能更好地落实公约，提出可行的预警机制实现。
